luzer/luzer_args.c

```diff
@@ -0,0 +1,255 @@
+		table_args->argv[table_args->argc] = arg;
+		table_args->argc++;
+	}
+	lua_pop(L, 1);
+	return 0;
+}
+
+NO_SANITIZE static int
+merge_args(luzer_args *cli_args, luzer_args *table_args, luzer_args *total_args) {
+	total_args->argc = (cli_args->argc + table_args->argc);
+	total_args->argv = malloc(sizeof(char*) * (total_args->argc + 1));
+	if (!cli_args->argv) {
+                return -1;
+        }
+
+	/* Program name on zero index. */
+	total_args->argv[0] = cli_args->argv[0];
+
+	int cur_pos_arg = 1;
+	char *corpus_path = NULL;
+	for (int i = 0; i < table_args->argc; i++) {
+		if (strncmp(table_args->argv[i], SEED_CORPUS_PATH_FLAG, strlen(SEED_CORPUS_PATH_FLAG)) == 0) {
+			int corpus_path_len = strlen(table_args->argv[i]) - strlen(SEED_CORPUS_PATH_FLAG);
+			corpus_path = malloc(corpus_path_len * sizeof(char*));
+                        if (!corpus_path) {
+                                return -1;
+                        }
+			memcpy(corpus_path, &table_args->argv[i][strlen(SEED_CORPUS_PATH_FLAG) + 1], corpus_path_len);
+			free(table_args->argv[i]);
+                        table_args->argv[i] = NULL;
+		} else {
+			total_args->argv[cur_pos_arg] = table_args->argv[i];
+			cur_pos_arg++;
+		}
+	}
+	for (int i = 1; i < cli_args->argc; i++) {
+		total_args->argv[cur_pos_arg] = cli_args->argv[i];
+		cur_pos_arg++;
+	}
+	if (corpus_path) {
+		total_args->argv[cur_pos_arg] = corpus_path;
+		cur_pos_arg++;
+	}
+	total_args->argv[total_args->argc] = NULL;
+
+        if (table_args->argv) {
+                free(table_args->argv);
+        }
+
+        if (cli_args->argv) {
+                free(cli_args->argv);
+        }
+
+	return 0;
+}
+
+NO_SANITIZE static void
+free_args(luzer_args args) {
+        if (!args.argv) {
+                return;
+        }
+        for (int i = 0; i < args.argc; i++) {
+                if (args.argv[i]) {
+                        free(args.argv[i]);
+                }
+        }
+        free(args.argv);
+}
+
+#ifdef DEBUG
+NO_SANITIZE static void
+print_args_with_prefix(luzer_args args, const char* prefix) {
+        for (int i = 0; i < args.argc; i++) {
+                DEBUG_PRINT("libFuzzer %s arg - '%s'\n", prefix, args.argv[i]);
+        }
+}
+#endif /* DEBUG */
+
+NO_SANITIZE int
+luaL_get_fuzz_args(lua_State *L, char ***argv, int *argc) {
+        luzer_args total_args = { .argv = NULL, .argc = 0};
+        luzer_args cli_args = { .argv = NULL, .argc = 0 };
+	luzer_args table_args = { .argv = NULL, .argc = 0 };
+
+        int result = -1;
+        result = luaL_get_args_from_cli(L, &cli_args);
+        if (result != 0) {
+                free_args(cli_args);
+                luaL_error(L, "failed parsing fuzz args. not enough memory");
+        }
+
+#ifdef DEBUG
+        print_args_with_prefix(cli_args, "from cli");
+#endif /* DEBUG */
+
+	/* If flag in cli and lua is duplicated, then flag from lua is ignored. */
+	result = luaL_get_args_from_table(L, &table_args, &cli_args);
```

Comment:
it is not clear why `cli_args` is passed to a function that gets args from a Lua table.

Reply:
Good idea, corrected